### PR TITLE
Fix tests for numpy 2.0

### PR DESCRIPTION
**Pull request by e-koch**
Test fixes against numpy 2.0

### CODE REVIEW

**Comment by e-koch** [comment]
Failures resulting from `ndarray.ptp` being removed in numpy >2.0 (see https://github.com/astropy/astropy/commit/fbf7c89725f34d92a9dcf739e2bea4371add7fd0)

**Comment by e-koch** [comment]
Removing ptp as a method fixed this issue. 

Remaining issues are related to the casadata package. @keflavich the fix that was working in the uvcombine doesn't seem to be working anymore

**Comment by e-koch** [comment]
Updating the data manually via casaconfig seems to fix the casadata path error: https://casadocs.readthedocs.io/en/stable/notebooks/external-data.html#Populating-the-Data-Directory-Manually

File: spectral_cube/io/casa_image.py [hunk 1]
```diff
@@ -105,6 +105,9 @@ def load_casa_image(filename, skipdata=False, memmap=True,
     elif 'beams' in beam_:
         bdict = beam_['beams']
 
+        # NOTE: temp to check failing test
+        print(desc['_keywords_']['coords'])
```
Comment (keflavich) [inline spectral_cube/io/casa_image.py:109]: just commenting as a reminder since this is a temp check

Comment (e-koch) [reply spectral_cube/io/casa_image.py:109]: thanks. still trying to figure out why the dict structure seems to have changed.

**Comment by e-koch** [comment]
@keflavich @preshanth the tests are hitting 2 failures that _seem_ to imply underlying changes in the image tool? I'm somewhat lost on where the change actually occurred.

The current failures have 2 sources:
1. https://github.com/radio-astro-tools/spectral-cube/actions/runs/10256840948/job/28376751741?pr=917#step:6:933 - Creation of the multi-beam test casa image is failing to set the restoring beam: https://github.com/radio-astro-tools/spectral-cube/actions/runs/10256840948/job/28376751741?pr=917#step:6:224
2. The `desc` dict that casa-formats-io returns has a different key structure: https://github.com/radio-astro-tools/spectral-cube/actions/runs/10256840948/job/28376751741?pr=917#step:6:720. The list of keys the test is returning right now is here: https://github.com/radio-astro-tools/spectral-cube/actions/runs/10256840948/job/28376751741?pr=917#step:6:727

**Comment by e-koch** [comment]
also note that the CASA builds are pinned to numpy<2.0.

There were a bunch of additional errors related to a build numpy 1.X build somewhere in the casa module stack. That seems firmly upstream of handling in spectral-cube.

**Comment by preshanth** [comment]
On casa numpy issue. There is ongoing work to fix casa and put out an updated version. That should fix the numpy 2.0 problems. Will look at the dict when I get a chabce

**Review by keflavich — APPROVED** [review]
can we go ahead and merge this and open a new issue/new PR for additional fixes?

**Comment by keflavich** [comment]
I'm gonna do what I suggested - being incompatible with numpy 2.0 is not ok, and no other PR tests are going to work without this.

**Pull request merged**